modules/compiler/utils/source/barrier_regions.cpp

```diff
@@ -264,6 +264,120 @@ void UpdateAndTrimPHINodeEdges(BasicBlock *BB, ValueToValueMapTy &vmap) {
+  if (gep_it != live_GEPs.end()) {
+    return gep_it->second;
+  }
+
+  Value *gep;
+  Type *data_ty = live->getType();
+  if (auto *AI = dyn_cast<AllocaInst>(live)) {
+    data_ty = AI->getAllocatedType();
+  }
+
+  if (!isa<ScalableVectorType>(data_ty)) {
+    auto field_it = barrier.live_variable_index_map_.find(live);
+    if (field_it == barrier.live_variable_index_map_.end()) {
+      return nullptr;
+    }
+
+    LLVMContext &context = barrier.module_.getContext();
+    unsigned field_index = field_it->second;
+    Value *live_variable_info_idxs[2] = {
+        ConstantInt::get(Type::getInt32Ty(context), 0),
+        ConstantInt::get(Type::getInt32Ty(context), field_index)};
+
+    gep = GetElementPtrInst::Create(
+        barrier.live_var_mem_ty_, barrier_struct, live_variable_info_idxs,
+        Twine("live_gep_") + live->getName(), insert_point);
+  } else {
+    auto field_it = barrier.live_variable_scalables_map_.find(live);
+    if (field_it == barrier.live_variable_scalables_map_.end()) {
+      return nullptr;
+    }
+    unsigned const field_offset = field_it->second;
+    Value *scaled_offset = nullptr;
+
+    LLVMContext &context = barrier.module_.getContext();
+    IRBuilder<> B(insert_point);
+    if (field_offset != 0) {
+      if (!vscale) {
+        Type *size_type = B.getIntNTy(barrier.size_t_bytes * 8);
+        vscale = B.CreateIntrinsic(Intrinsic::vscale, size_type, {});
+      }
+      scaled_offset = B.CreateMul(
+          vscale, B.getIntN(barrier.size_t_bytes * 8, field_offset));
+    } else {
+      scaled_offset = ConstantInt::get(Type::getInt32Ty(context), 0);
+    }
+
+    Value *live_variable_info_idxs[3] = {
+        ConstantInt::get(Type::getInt32Ty(context), 0),
+        ConstantInt::get(Type::getInt32Ty(context),
+                         barrier.live_var_mem_scalables_index),
+        scaled_offset,
+    };
+
+    // Gep into the raw byte buffer
+    gep = B.CreateInBoundsGEP(barrier.live_var_mem_ty_, barrier_struct,
+                              live_variable_info_idxs,
+                              Twine("live_gep_scalable_") + live->getName());
+
+    // Cast the pointer to the scalable vector type
+    gep = B.CreatePointerCast(
+        gep,
+        PointerType::get(
+            data_ty,
+            cast<PointerType>(barrier_struct->getType())->getAddressSpace()));
+  }
+
+  live_GEPs.insert(std::make_pair(live, gep));
+  return gep;
+}
+
+Value *compiler::utils::Barrier::LiveValuesHelper::getReload(
+    Value *live, Instruction *insert, const char *name, bool reuse) {
+  auto &mapped = reloads[live];
+  if (reuse && mapped) {
+    return mapped;
+  }
+
+  Value *v = getGEP(live);
+  if (v) {
+    if (!isa<AllocaInst>(live)) {
+      // If live variable is not allocainst, insert load.
+      v = new LoadInst(live->getType(), v, Twine(live->getName(), name),
+                       insert);
+    }
+    mapped = v;
+    return v;
+  } else if (auto *I = dyn_cast<Instruction>(live)) {
```

Comment:
can maybe remove `else` after `return` - your call

Reply:
yeah doesn't really make much difference imho other than forcing the `if` onto the next line